spec/tlc/traffic_light_controller_proxy_integration_spec.rb

```diff
@@ -0,0 +1,152 @@
+RSpec.describe 'TLC Proxy Integration' do
+  let(:timeout) { 1 }
+  let(:ip) { 'localhost' }
+  let(:port) { 13112 }  # Different port to avoid conflicts
+  let(:site_id) { 'TLC001' }
+  
+  let(:tlc_site_settings) {
+    {
+      'site_id' => site_id,
+      'supervisors' => [
+        { 'ip' => ip, 'port' => port }
+      ],
+      'sxl' => 'tlc',
+      'sxl_version' => '1.0.13',
+      'components' => {
+        site_id => { 'type' => 'main' }
+      },
+      'signal_plans' => {
+        1 => { 'cycle_time' => 60 },
+        2 => { 'cycle_time' => 80 },
+        3 => { 'cycle_time' => 120 }
+      },
+      'security_codes' => { 1 => '1111', 2 => '2222' }
+    }
+  }
+  
+  let(:supervisor_settings) {
+    {
+      'port' => port,
+      'sites' => {
+        site_id => { 'sxl' => 'tlc', 'type' => 'tlc' }
+      },
+      'guest' => {
+        'sxl' => 'tlc',  # Use valid schema type
+        'type' => 'tlc'   # Proxy type
+      }
+    }
+  }
+  
+  let(:log_settings) {
+    {
+      'active' => false
+    }
+  }
+  
+  let(:tlc_site) {
+    RSMP::TLC::TrafficControllerSite.new(
+      site_settings: tlc_site_settings,
+      log_settings: log_settings
+    )
+  }
+  
+  let(:supervisor) {
+    RSMP::Supervisor.new(
+      supervisor_settings: supervisor_settings,
+      log_settings: log_settings
+    )
+  }
+
+  it 'creates TLCProxy when TLC connects to supervisor' do
+    expect(supervisor.proxies.size).to eq(0)
+    expect(tlc_site.proxies.size).to eq(1)
+    expect(tlc_site.proxies.first.state).to eq(:disconnected)
+
+    AsyncRSpec.async context: lambda {
+      supervisor.start
+      tlc_site.start
+    } do |task|
+      # Wait for TLC site to connect to supervisor
+      tlc_proxy = supervisor.wait_for_site site_id, timeout: timeout
+      supervisor_proxy = tlc_site.wait_for_supervisor ip, timeout: timeout
+      
+      # Verify that supervisor created a TLCProxy instead of regular SiteProxy
```

Comment:
Need to be consistent with the naming, the class is not called TLCProxy, but TrafficLightControllerSite (and i suggested elsewhere to to TrafficControllerProxy). I've seen TLCProxy used elsewhere too.